### PR TITLE
Add JCSDA packages solo, r2d2, ewok; constrain yafyaml version to 0.5.1 for mapl 2.x.y, add jedi-mpas-env

### LINKED ISSUE #164: Add remaining packages required for EWOK experiments to jedi-ewok-env
These packages are still missing:
- solo
- r2d2
- ewok

They are currently not available to the public, therefore add as variants (off by default) to jedi-ewok-env

### PR DESCRIPTION
**Pull request by climbfuji**
- Add JCSDA packages `solo`, `r2d2`, `ewok`. These Python packages are currently available for internal developers only, therefore the corresponding variants in `jedi-ewok-env` are off by default.
- Constrain `yafyaml` version to `0.5.1` for `mapl-2.x.y` as instructed by the developers.
- Add `jedi-mpas-env`: for now, only builds what is in `jedi-base-env`, will add any missing packages that I am not aware of later.

Tested on my macOS with `clang-13.1.6` and `openmpi-4.1.3`. Note that the CI tests will not attempt to build `solo`, `r2d2`, `ewok`.

Fixes https://github.com/NOAA-EMC/spack-stack/issues/164

### CODE REVIEW

File: var/spack/repos/jcsda-emc-bundles/packages/jedi-mpas-env/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    git      = "https://github.com/JCSDA/mpas-bundle.git"
+
+    maintainers = ['climbfuji', 'rhoneyager']
+
+    version('main')
+
+    depends_on('base-env',          type='run')
+    depends_on('jedi-base-env',     type='run')
+
+    # Anything missing?
```
Comment (kgerheiser) [inline var/spack/repos/jcsda-emc-bundles/packages/jedi-mpas-env/package.py]: Missing a newline :)

Comment (kgerheiser) [inline var/spack/repos/jcsda-emc-bundles/packages/jedi-mpas-env/package.py]: Missing a newline :)

**Review by kgerheiser — APPROVED** [review]

File: var/spack/repos/jcsda-emc-bundles/packages/jedi-mpas-env/package.py (hunk 1 above)
Comment (climbfuji) [inline var/spack/repos/jcsda-emc-bundles/packages/jedi-mpas-env/package.py]:
```suggestion
    # Anything missing?

```

Comment (climbfuji) [reply var/spack/repos/jcsda-emc-bundles/packages/jedi-mpas-env/package.py]: Of course! Fixed :-)

**Pull request merged**